src/coreclr/src/jit/gentree.h

```diff
@@ -4727,21 +4727,36 @@ struct GenTreeQmark : public GenTreeOp
 struct GenTreeIntrinsic : public GenTreeOp
 {
     CorInfoIntrinsics     gtIntrinsicId;
+    NamedIntrinsic        gtIntrinsicName;
```

Comment:
How is `gtIntrinsicName` vs `gtIntrinsicId` being differentiated? Just by one of them being invalid?

Reply:
Good point. I am open for suggestions here. :)
Currently call sights are roughly doing `gtIntrinsicId` xor `gtIntrinsicName`. All the math functions are converted to QCalls and their corresponding intrinsics to NamedIntrinsic. I think the long-term plan is to get rid of most, if not all, the FCalls in CoreCLR. If that happens, `gIntrinsicId` will go away in the future.

Reply:
I was mostly just interested here. I think, longer term, it might be good to make `GenTreeJitIntrinsic` the general purpose holder for `NamedIntrinsic`, but that is a bigger change and probably needs more discussion/is out of scope for .NET 5.